engine/execution/pruner/core.go

```diff
@@ -155,3 +164,18 @@ func makeBlockIteratorCreator(
 		return next, header.Height, nil
 	}, nil
 }
+
+// estimateBatchProcessing estimates the number of batches and the total duration
+func EstimateBatchProcessing(
+	start, end uint64, batchSize uint, sleepAfterEachBatchCommit time.Duration, commitDuration time.Duration) (
+	batchCount uint64, totalDuration time.Duration) {
+	if batchSize == 0 || start >= end {
```

Comment:
Because `end` is inclusive, `start == end` is valid (with one block to iterate).

```suggestion
	if batchSize == 0 || start > end {
```